### PR TITLE
Add the intitial PeerManager feature doc

### PR DESCRIPTION
**Pull request by agunde406**
The feature doc is currently made up of a high level
description, a dependency graph, and sequence diagrams
for connecting/reconnecting to peers.

This doc will be expanded in the future.

### CODE REVIEW

File: community/planning/peer_manager.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+<!--
+  Copyright 2018-2020 Cargill Incorporated
+  Licensed under Creative Commons Attribution 4.0 International License
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+## Summary
+The `PeerManager` is the component of the system that knows about peers. It
+has a `Connector` to the `ConnectionManager` and requests the creation and
+removal of connection that are associated with a peer. The `PeerManager`
```
Comment (peterschwarz) [inline community/planning/peer_manager.md:11]: s/connection/connections

File: community/planning/peer_manager.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ID, a list of endpoints and which endpoint is currently active.
+
+## Dependencies
+
+![]({% link images/peer_manager_dep.svg %} "PeerManager Dependency Diagram")
+
+## Sequence Diagrams
+
+### Connecting to a new Peer
+The `PeerManagerConnector` can be used by other components to request connection
```
Comment (peterschwarz) [inline community/planning/peer_manager.md:22]: s/request connection/request a connection

File: community/planning/peer_manager.md (hunk 1 above)
Comment (agunde406) [reply community/planning/peer_manager.md]: fixed

File: community/planning/peer_manager.md (hunk 2 above)
Comment (agunde406) [reply community/planning/peer_manager.md]: fixed

**Pull request merged**